docs/extend/extend-cli/cli-implement-profiles.md

```diff
@@ -1,12 +1,12 @@
 # Implementing profiles in a plug-in
-You can use this profile template to create a profile for your product.  
 
-The profile definition is placed in the `imperative.ts` file.
+Create a profile template that you can apply to create a profile for your product.  
 
-The `type: "someproduct"` property represents the profile name that you might require on various commands to have credentials loaded 
-from a secure credential manager and retain the host/port information, so that you can easily swap to different servers from the CLI.
+The profile definition is placed in the `imperative.ts` file. **[what is the "profile definition" here? are we saying that this code should be added to the imperative file?]**
```

Comment:
@adam-wolfe please respond

Reply:
For a plug-in that connects to a service other than the ones that are supported in Zowe CLI by default (z/OSMF, TSO, and SSH), they must define a new profile type and define the properties that can be specified (in a team config) for the new profile type.

For a plug-in to define new profile types, they would add a `profiles` array in the plug-in's `config` object (in `imperative.ts`) that looks like the example below.

It might be good in the future to add a bit more detail to the example, since it doesn't provide the full context.

Reply:
discussed in person